app/Commands/UploadTrackCommand.php

```diff
@@ -147,13 +153,16 @@ public function execute()
                 Log::info('Encoding ' . $track->id . ' into ' . $target);
                 $this->notify('Encoding ' . $name, $index / count(Track::$Formats) * 100);
 
-                $pipes = [];
-                $proc = proc_open($command, [0 => ['pipe', 'r'], 1 => ['pipe', 'w'], 2 => ['pipe', 'a']], $pipes);
-                $processes[] = $proc;
-            }
+                $process = new Process($command);
+                $process->mustRun();
+
+                // Update file size for track file
+                $trackFile->updateFilesize();
 
-            foreach ($processes as $proc) {
-                proc_close($proc);
+                // Delete track file if it is cacheable
+                if ($trackFile->is_cacheable == true) {
+                    \Illuminate\Support\Facades\File::delete($trackFile->getFile());
```

Comment:
Right. That class looks redundant to me, it provides the same functionality as `Response::download`.